lib/dids/key_test.go

```diff
@@ -116,8 +120,44 @@ func TestBasicSignVerify(t *testing.T) {
 	assert.True(t, valid)
 
 	// create modified/incorrect block with different content
-	modifiedBlock := blocks.NewBlock([]byte("modified data 123 456"))
+	modifiedBlock := blocks.NewBlock([]byte("data doesn't matter"))
 	valid, err = did.Verify(modifiedBlock, jws1)
 	assert.NotNil(t, err)
 	assert.False(t, valid)
 }
+func TestRealIpfsVerify(t *testing.T) {
+	// parse the real provided CID
+	block, err := blocks.NewBlockWithCid([]byte("data doesn't matter"), cid.MustParse("bafyreiapoy2k666lcpsnbbilyzloqbc64mevrkymorujyi4cjqkzp2ukj4"))
+	assert.Nil(t, err)
+
+	// the real DID from the transaction
+	realKeyDid := "did:key:z6MkoqsapCm3eQXQBByJTFcL5SMFg7d3aH8iobtmoL8tt6yw"
+
+	// the real sig from the transaction
+	realSig := "SV1jI7m-Fs59cBioCxrV1oLIbsbA-nDAQtpFUFDvr2iAxeq5JnfUHb2N8z2N78Lt4PnsMeX9flaq-IpzN97wDA"
+
+	// create the header
+	header := map[string]interface{}{
+		"alg": "EdDSA",
+		"kid": realKeyDid + "#" + strings.Split(realKeyDid, ":")[2],
+	}
+
+	// marshal the header and encode in base64-url
+	headerJSON, err := json.Marshal(header)
+	assert.Nil(t, err)
+	encodedHeader := base64.RawURLEncoding.EncodeToString(headerJSON)
+
+	// encode the CID (payload) in base64-url
+	encodedPayload := base64.RawURLEncoding.EncodeToString(block.Cid().Bytes())
+
+	// recreate the full signature string in JWT format (header.payload.signature)
+	fullSignature := encodedHeader + "." + encodedPayload + "." + realSig
```

Comment:
There shouldn't be all this boilerplate to verify a signature. `realSig` should just be passed in directly, and when signing a message, that is what should be returned.